### PR TITLE
assignment 1 (plus challenge 1 and 3)

### PR DESCRIPTION
**Pull request by claravania**

### CODE REVIEW

File: claravania/assignment-1/anagrams.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    itemSet2 = defaultdict(int)
+
+    # if both strings have different length return False
+    if len(str1) != len(str2):
+        return False
+
+    if case == "insensitive":
+        str1, str2 = str1.lower(), str2.lower()
+
+    for item1, item2 in zip(str1, str2):
```
Comment (kevmo314) [inline claravania/assignment-1/anagrams.py]: Does this work if you call `check_anagrams('abc', 'abcd')`? `zip()` won't work too well for strings of different length here.

File: claravania/assignment-1/anagrams.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    Similar to challenge 1, we handle both case sensitive and insensitive cases.
+    """
+
+    word1 = sent1.split()[0]
+    word2 = sent2.split()[0]
+
+    return check_anagrams(word1, word2, case)
+
+
+def main():
```
Comment (kevmo314) [inline claravania/assignment-1/anagrams.py]: You don't have to for this assignment, but you may want to look into using `unittest` or similar in the future :)

https://docs.python.org/3/library/unittest.html

File: claravania/assignment-1/anagrams.py (hunk 1 above)
Comment (claravania) [reply claravania/assignment-1/anagrams.py]: I checked first whether the two strings have the same length or not, so I suppose both cases would be fine in this case?

https://github.com/Google-CodeU-EMEA/Group-7/blob/3f4c1f7528d79508948b58a4f9d4b97cd73c132f/claravania/assignment-1/anagrams.py#L16

File: claravania/assignment-1/anagrams.py (hunk 2 above)
Comment (claravania) [reply claravania/assignment-1/anagrams.py]: Thanks for the pointer!

File: claravania/assignment-1/anagrams.py (hunk 1 above)
Comment (kevmo314) [reply claravania/assignment-1/anagrams.py]: Ah yes I see, nice. :)

**Comment by kevmo314** [comment]
Click that merge pull request button when you're ready :D

**Pull request merged**